language/stdlib/modules/compare.move

```diff
@@ -0,0 +1,99 @@
+// This defines a total order `Compare::cmp<T>` among Move values of the same type `T`. That is, we
+// have the guarantee:
+// `Compare::cmp<T>(t1, t2) == LESS_THEN` iff `Compare::cmp<T>(t2, t1) == GREATER_THEN`
+//
+// Move has bytecodes for polymorphic equality `==` and inequality `!=`. The implementation
+// of `Compare::cmp<T>(t1, t2)` also guarantees that
+// `Compare::cmp<T>(t1, t2) == EQUAL` iff `t1 == t2` and (similarly)
+// `Compare::cmp<T>(t1, t2) != EQUAL` iff `t1 != t2`.
+//
+// We implement `Compare.cmp(t1, t2)` using the `vector<u8>`'s produced by `LCS::to_bytes(t1)` and
+// `LCS::to_bytes(t2)`. We compare the vectors first by length, then byte-by-byte from right to left
+// if the lengths are equal. The length comparison is done in the obvious way; we return
+// `LESS_THAN` if len(lcs(t1)) < len(lcs(t2)) and `GREATER_THAN` if len(lcs(t1)) > len(lcs(t2)).
+// This ensures that for all primitive types `T` with `<` and `>` comparison operators exposed in
+// Move bytecode (`u8`, `u64`, `u128`), we have
+// `Compare::cmp<T>(t1, t2) == LESS_THEN` iff `t1 < t2` and (similarly)
+// `Compare::cmp<T>(t1, t2) == GREATER_THEN` iff `t1 > t2`.
```

Comment:
Thoughts on source language syntactic sugar that translates `t1 < t2` to `Compare::cmp<T>(t1, t2) == LESS_THEN`